packages/ocs/dashboards/persistent-internal/encryption-popover.tsx

```diff
@@ -0,0 +1,235 @@
+  }
+};
+
+const renderEncryptionDetails = (
+  t: TFunction,
+  kmsTypeDescription: string,
+  encryptionDetails: EncryptionPopoverProps['encryptionDetails'],
+  hasStorageClassEncryption: boolean,
+  hasMCGEncryption: boolean,
+  isKmsEnabled: boolean,
+  inTransitEncryption: boolean
+) => {
+  return (
+    <TextContent>
+      {hasMCGEncryption ? (
+        <>
+          <Flex alignItems={{ default: 'alignItemsCenter' }}>
+            <FlexItem>
+              <Text component={TextVariants.h6}>{t('Object storage')}</Text>
+            </FlexItem>
+            <FlexItem align={{ default: 'alignRight' }}>
+              <StatusIcon enabled={hasMCGEncryption} />
+            </FlexItem>
+          </Flex>
+          <Text>
+            {t('Data encryption for object storage.')}
+            {kmsTypeDescription && (
+              <Text component={TextVariants.small}>{kmsTypeDescription}</Text>
+            )}
+          </Text>
+        </>
+      ) : (
+        <>
+          <Flex alignItems={{ default: 'alignItemsCenter' }}>
+            <FlexItem>
+              <Text component={TextVariants.h6}>
+                {t('Cluster-wide encryption')}
+              </Text>
+            </FlexItem>
+            <FlexItem align={{ default: 'alignRight' }}>
+              <StatusIcon enabled={encryptionDetails.clusterWide} />
+            </FlexItem>
+          </Flex>
+          <Text>
+            {t('Encryption for the entire cluster (block and file)')}
+            <br />
+            {kmsTypeDescription && (
+              <Text component={TextVariants.small}>{kmsTypeDescription}</Text>
+            )}
+          </Text>
+
+          {hasStorageClassEncryption && (
+            <>
+              <Divider />
+              <Text component={TextVariants.h6}>
+                {t('Storage class encryption')}
+              </Text>
+              <Text>
+                {t('Encryption for PVs')}
+                {kmsTypeDescription && (
+                  <Text component={TextVariants.small}>
+                    {kmsTypeDescription}
+                  </Text>
+                )}
+              </Text>
+
+              <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                <FlexItem>
+                  <Text component={TextVariants.h6}>{t('Block storage')}</Text>
+                </FlexItem>
+                <FlexItem align={{ default: 'alignRight' }}>
+                  <StatusIcon
+                    enabled={isKmsEnabled && hasStorageClassEncryption}
+                  />
+                </FlexItem>
+              </Flex>
+              <Text>{t('Data encryption for block storage.')}</Text>
+
+              <Flex alignItems={{ default: 'alignItemsCenter' }}>
+                <FlexItem>
+                  <Text component={TextVariants.h6}>{t('File storage')}</Text>
+                </FlexItem>
+                <FlexItem align={{ default: 'alignRight' }}>
+                  <StatusIcon enabled={encryptionDetails.clusterWide} />
```

Comment:
this is wrong, why are u using clusterwide flag for filesystem encrypted storageclass ??
```suggestion
                  <Text component={TextVariants.h6}>{t('File storage')}</Text>
                </FlexItem>
                <FlexItem align={{ default: 'alignRight' }}>
                  <StatusIcon enabled={} />
```